### DISCUSSION TITLE
New "wait time before cure" tool

### DISCUSSION THREAD

**Discussion by bryanseger** [sn4k3/UVtools#481] [Ideas]
Similar to the Exposure Time Finder tool, I'd like to see a "wait time before cure" tool to help battle the effects of blooming.

This thought occurred as I was running an exposure test actually. I was printing a grid of several different tests, all .1s apart from each other, and the quickest piece had nasty blooming, and the next quickest had a subtle amount, and the rest had nearly zero blooming. 

Looking at the by-layer data, the quickest piece prints directly after the print head moves a full up/down, whereas all other pieces print directly after just a small .1mm head movement. 

So, without editing every layer manually, could we get a simple test that, say, prints X number of cylinders, but each cylinder has a different "wait before cure" time prior to cure. Between each cure iteration of each cylinder, the print head would need to cycle a full up/down movement to mimic true movement of a normal print. Hopefully that makes sense.

**Comment by sn4k3** [comment]
Yes, it's a good ideia. I will write it.

**Reply by bryanseger** [reply]
Tiago,

I just tried the "blooming test" for the first time (I know, I waited nearly 2 years to actually try it...). I'm working on a Phrozen Mega 8KS printer. I set my bottom exp time to 10s and the normal exp time to 0.9s, but the blooming test makes every single layer at the 10s exp time. Is this an issue in the blooming test creation, or an issue on the printer side?

I have attached an example file for reference. It's an easy fix to get it to run correctly, but I wanted to mention it to you in case it's an easy fix for your next update.
[megacubev4 test file.zip](https://github.com/sn4k3/UVtools/files/14298978/megacubev4.test.file.zip)

**Reply by sn4k3** [reply]
Will be fixed on next release.
Meanwhile if you want to fix it you can run this code under the UVtools terminal (Ctrl + Shift + T):

```C#
foreach (var layer in SlicerFile)
{
    if (layer.IsBottomLayerByHeight) continue;
    layer.ExposureTime = SlicerFile.ExposureTime;
}
```

Then you need to save the file.

**Reply by bryanseger** [reply]
Awesome, thanks Tiago!
